airflow/io/path.py

```diff
@@ -123,20 +120,14 @@ def __new__(cls: type[PT], *args: str | os.PathLike, **kwargs: typing.Any) -> PT
 
         url = stringify_path(other)
         parsed_url: SplitResult = urlsplit(url)
-        protocol: str | None = split_protocol(url)[0] or parsed_url.scheme
-
-        # allow override of protocol
-        protocol = kwargs.get("scheme", protocol)
 
-        for key in ["scheme", "url"]:
-            val = kwargs.pop(key, None)
-            if val:
-                parsed_url = parsed_url._replace(**{key: val})
+        if scheme:  # allow override of protocol
+            parsed_url = parsed_url._replace(scheme=scheme)
```

Comment:
I removed `url` since it does not work:

```pycon
>>> from urllib.parse import urlsplit
>>> urlsplit("s3://foo/bar")._replace(url="")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/uranusjr/Library/PythonUp/versions/3.12/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/collections/__init__.py", line 460, in _replace
    raise ValueError(f'Got unexpected field names: {list(kwds)!r}')
ValueError: Got unexpected field names: ['url']
```

Let me know if it represents some intention without a correct implementation so I can add the logic back.

Reply:
Not really. This is ported from upath and made to look the same as much as possible. So I haven't looked at all the angles.